### PR TITLE
Recipe App

### PR DESCRIPTION
**Pull request by AlexRS90**
For this group project week, we built a Recipe App that let the user keep track of his own recipes and also from the other user when the recipe is public he can also generate a shopping list that will include all the missing foods from his inventory. The app includes the following features: ⬇️  ⬇️ 

- A login page.
- A registration page.
- A food list (with all CRUD implementation, except for 'update').
- A recipes list (with all CRUD implementation, except for 'update').
- A general shopping list view (all missing food for all your recipes and total price).
- A list of all public recipes from other users with their names and total prices.

Each feature lets the user do: ⬇️  ⬇️ 

- Login page and registration page:
> Should be built with Devise.

- Food list:
> Should display a list of food as in the wireframe.
> Should lead to the form that allows users to add new food.

- Recipes list:
> Should display a list of recipes created by the logged-in user as in the wireframe.
> Should lead to recipe details.
> If the user is the owner of the recipe, should allow the user to delete it.

- Public recipe list:
> Should display a list of all public recipes ordered by newest as in the wireframe.
> Should lead to recipe details.

- Recipe details:
> If the recipe is public or the user is the owner of the recipe, should display the recipe details as in the wireframe.
> If the user is the owner of the recipe, should lead to the form that allows the user to add new food.

- Inventory:
> The user is able to create a new inventory to know how much stock he has.
> The user can add food to its inventory

- General shopping list view:
> Should show the list of food that is missing for all recipes of the logged-in user.
> Should count the total food items and total price of the missing food.
> Only can generate it if an inventory exists and if the recipe has its foods as ingredients.

- There are no N+1 queries happening.
- Created a navigation menu that allows users to open all of the pages you created.

Cooking time!!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+Contributions, issues, and feature requests are welcome!
+
+
+## Show your support
+
+Give a ⭐️ if you like this project!
+
+## 📝 License
+
+This project is [MIT](./MIT.md) licensed.
```
Comment (MahmoudBakr23) [inline README.md:80]: - [ ] Kindly fix the broken MIT link for a completely professional README file and repo 🙏 😄

File: app/views/inventory/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        </div>
+      </div>
+  </div>
+<% else %>
+  <%= form_with(model: @inventory, url: '/inventory/create', scope: :data) do |f| %>
+    <p>Looks like you don't have an inventory, let's create a new one =D</p>
+    <%# f.label(:name, 'Name:') %>
+    <%= f.text_field :name, class: "form-control", placeholder: 'How do you wanna name me?' %> <br>
+    <%= f.submit 'Save Inventory', class: "btn btn-primary" %>
+  <% end %>
```
Comment (MahmoudBakr23) [inline app/views/inventory/index.html.erb:21]: - [ ] Nice job implementing this logic so far in order to serve the need of your app. However, it's always nicer, safer, and cleaner to separate the logic from the views and move it to the rails helpers (I mean, that's what the helpers exist for right?) 😉

**Review by MahmoudBakr23 — APPROVED** [review]
Hi team😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉 

- Linters are correct
- Professional PR headline and description
- Great use of RSpec to test the integrations and units of your app
- Great routing setup
- Well done using partials in your views
- Well done using the correct methods to solve the n+1 query problems and reduce the number of queries requested from the controllers
- Great design

**OPTIONAL**
_Kindly check the comments down below._ 🙏 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/inventory/index.html.erb (hunk 2 above)
Comment (AlexRS90) [reply app/views/inventory/index.html.erb:21]: Thank you very much for your suggestion! and yes you're right! 🚀

**Pull request merged**